src/udev/udev-builtin-blkid.c

```diff
@@ -244,13 +245,16 @@ static int builtin_blkid(struct udev_device *dev, int argc, char *argv[], bool t
         for (;;) {
                 int option;
 
-                option = getopt_long(argc, argv, "oR", options, NULL);
+                option = getopt_long(argc, argv, "o::R", options, NULL);
                 if (option == -1)
                         break;
 
                 switch (option) {
                 case 'o':
-                        offset = strtoull(optarg, NULL, 0);
+                        if (optarg)
+                                err = safe_atoi64(optarg, &offset);
```

Comment:
Hmm, using safe_atoi64() is definitely the better choice given the type of `offset` and what blkid_probe_set_device() actually wants. However, I think we should add extra validation here to filter out negative offsets, since those don't actually make sense. i.e. please set err to -ERANGE if offset is noticed to be negative.